diff --git a/node_modules/expo-constants/android/build.gradle b/node_modules/expo-constants/android/build.gradle
index 7de3bcc..ffcf427 100644
--- a/node_modules/expo-constants/android/build.gradle
+++ b/node_modules/expo-constants/android/build.gradle
@@ -10,11 +10,11 @@ version = '57.0.6'
 
 expoModule {
   // We can't prebuild the module because we need to apply `get-app-config-android.gradle` script.
-  canBePublished false
+  canBePublished = false
 }
 
 android {
-  namespace "expo.modules.constants"
+  namespace = "expo.modules.constants"
   defaultConfig {
     versionCode 33
     versionName "57.0.6"
diff --git a/node_modules/expo-constants/android/src/main/java/expo/modules/constants/ConstantsModule.kt b/node_modules/expo-constants/android/src/main/java/expo/modules/constants/ConstantsModule.kt
index 67247f7..e504cc0 100644
--- a/node_modules/expo-constants/android/src/main/java/expo/modules/constants/ConstantsModule.kt
+++ b/node_modules/expo-constants/android/src/main/java/expo/modules/constants/ConstantsModule.kt
@@ -10,8 +10,9 @@ class ConstantsModule : Module() {
   override fun definition() = ModuleDefinition {
     Name("ExponentConstants")
 
-    Constants {
-      return@Constants appContext.service<ConstantsInterface>()?.constants ?: emptyMap()
+    val constants = appContext.service<ConstantsInterface>()?.constants ?: emptyMap()
+    constants.forEach { (name, value) ->
+      Constant(name) { value }
     }
 
     AsyncFunction<String?>("getWebViewUserAgentAsync") {
