diff --git a/node_modules/react-native-safe-area-context/android/build.gradle b/node_modules/react-native-safe-area-context/android/build.gradle
index f324b9a..85938ed 100644
--- a/node_modules/react-native-safe-area-context/android/build.gradle
+++ b/node_modules/react-native-safe-area-context/android/build.gradle
@@ -41,37 +41,38 @@ if (isNewArchitectureEnabled()) {
 android {
     def agpVersion = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION
     if (agpVersion.tokenize('.')[0].toInteger() >= 7) {
-        namespace "com.th3rdwave.safeareacontext"
+        namespace = "com.th3rdwave.safeareacontext"
         buildFeatures {
-            buildConfig true
+            buildConfig = true
         }
     }

-    compileSdkVersion getExtOrDefault('compileSdkVersion', 30)
+    compileSdk = getExtOrDefault('compileSdkVersion', 30)

     // Used to override the NDK path/version on internal CI or by allowing
     // users to customize the NDK path/version from their root project (e.g. for M1 support)
     if (rootProject.hasProperty("ndkPath")) {
-        ndkPath rootProject.ext.ndkPath
+        ndkPath = rootProject.ext.ndkPath
     }
     if (rootProject.hasProperty("ndkVersion")) {
-        ndkVersion rootProject.ext.ndkVersion
+        ndkVersion = rootProject.ext.ndkVersion
     }

     defaultConfig {
-        minSdkVersion getExtOrDefault('minSdkVersion', 16)
-        targetSdkVersion getExtOrDefault('targetSdkVersion', 28)
-        versionCode 1
-        versionName "1.0"
+        minSdk = getExtOrDefault('minSdkVersion', 16)
+        targetSdk = getExtOrDefault('targetSdkVersion', 28)
+        versionCode = 1
+        versionName = "1.0"
         buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()

         ndk {
             abiFilters (*reactNativeArchitectures())
         }
     }
-
-    lintOptions{
-        abortOnError false
+    lint {
+        abortOnError = true
+        warningsAsErrors = true
+        checkReleaseBuilds = true
     }

     packagingOptions {
@@ -111,7 +112,7 @@ repositories {
     google()
     maven {
         // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
-        url "$rootDir/../node_modules/react-native/android"
+        url = uri("$rootDir/../node_modules/react-native/android")
     }
     mavenCentral()
 }
@@ -119,6 +120,6 @@ repositories {
 def kotlin_version = getExtOrDefault('kotlinVersion', project.properties['RNSAC_kotlinVersion'])

 dependencies {
-    implementation 'com.facebook.react:react-native:+'
+    implementation 'com.facebook.react:react-android'
     implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
 }
diff --git a/node_modules/react-native-safe-area-context/android/src/main/AndroidManifest.xml b/node_modules/react-native-safe-area-context/android/src/main/AndroidManifest.xml
index f5da598..4b624d1 100644
--- a/node_modules/react-native-safe-area-context/android/src/main/AndroidManifest.xml
+++ b/node_modules/react-native-safe-area-context/android/src/main/AndroidManifest.xml
@@ -1,5 +1,4 @@

 <manifest
-	xmlns:android="http://schemas.android.com/apk/res/android"
-  package="com.th3rdwave.safeareacontext">
+	xmlns:android="http://schemas.android.com/apk/res/android">
 </manifest>
diff --git a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaProviderManager.kt b/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaProviderManager.kt
index de6f685..6f689f1 100644
--- a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaProviderManager.kt
+++ b/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaProviderManager.kt
@@ -36,6 +36,6 @@ class SafeAreaProviderManager :
 private fun handleOnInsetsChange(view: SafeAreaProvider, insets: EdgeInsets, frame: Rect) {
   val reactContext = view.context as ReactContext
   val reactTag = view.id
-  UIManagerHelper.getEventDispatcherForReactTag(reactContext, reactTag)
+  UIManagerHelper.getEventDispatcher(reactContext)
       ?.dispatchEvent(InsetsChangeEvent(getSurfaceId(reactContext), reactTag, insets, frame))
 }
diff --git a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaView.kt b/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaView.kt
index b29e787..ce03b37 100644
--- a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaView.kt
+++ b/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaView.kt
@@ -1,17 +1,11 @@
 package com.th3rdwave.safeareacontext

 import android.content.Context
-import android.util.Log
 import android.view.View
 import android.view.ViewTreeObserver
 import com.facebook.react.bridge.Arguments
 import com.facebook.react.uimanager.StateWrapper
-import com.facebook.react.uimanager.UIManagerModule
 import com.facebook.react.views.view.ReactViewGroup
-import java.util.concurrent.locks.ReentrantLock
-import kotlin.concurrent.withLock
-
-private const val MAX_WAIT_TIME_NANO = 500000000L // 500ms

 class SafeAreaView(context: Context?) :
     ReactViewGroup(context), ViewTreeObserver.OnPreDrawListener {
@@ -44,60 +38,8 @@ class SafeAreaView(context: Context?) :
         val map = Arguments.createMap()
         map.putMap("insets", edgeInsetsToJsMap(insets))
         stateWrapper.updateState(map)
-      } else {
-        val localData = SafeAreaViewLocalData(insets = insets, mode = mMode, edges = edges)
-        val reactContext = getReactContext(this)
-        val uiManager = reactContext.getNativeModule(UIManagerModule::class.java)
-        if (uiManager != null) {
-          uiManager.setViewLocalData(id, localData)
-          // Sadly there doesn't seem to be a way to properly dirty a yoga node from java, so if we
-          // are in
-          // the middle of a layout, we need to recompute it. There is also no way to know whether
-          // we
-          // are in the middle of a layout so always do it.
-          reactContext.runOnNativeModulesQueueThread {
-            uiManager.uiImplementation.dispatchViewUpdates(-1)
-          }
-          waitForReactLayout()
-        }
-      }
-    }
-  }
-
-  private fun waitForReactLayout() {
-    // Block the main thread until the native module thread is finished with
-    // its current tasks. To do this we use the done boolean as a lock and enqueue
-    // a task on the native modules thread. When the task runs we can unblock the
-    // main thread. This should be safe as long as the native modules thread
-    // does not block waiting on the main thread.
-    var done = false
-    val lock = ReentrantLock()
-    val condition = lock.newCondition()
-    val startTime = System.nanoTime()
-    var waitTime = 0L
-    getReactContext(this).runOnNativeModulesQueueThread {
-      lock.withLock {
-        if (!done) {
-          done = true
-          condition.signal()
-        }
-      }
-    }
-    lock.withLock {
-      while (!done && waitTime < MAX_WAIT_TIME_NANO) {
-        try {
-          condition.awaitNanos(MAX_WAIT_TIME_NANO)
-        } catch (ex: InterruptedException) {
-          // In case of an interrupt just give up waiting.
-          done = true
-        }
-        waitTime += System.nanoTime() - startTime
       }
     }
-    // Timed out waiting.
-    if (waitTime >= MAX_WAIT_TIME_NANO) {
-      Log.w("SafeAreaView", "Timed out waiting for layout.")
-    }
   }

   fun setMode(mode: SafeAreaViewMode) {
diff --git a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewManager.kt b/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewManager.kt
index 373c9dd..5f806cb 100644
--- a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewManager.kt
+++ b/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewManager.kt
@@ -15,10 +15,6 @@ class SafeAreaViewManager : ReactViewManager() {

   override fun createViewInstance(context: ThemedReactContext) = SafeAreaView(context)

-  override fun createShadowNodeInstance() = SafeAreaViewShadowNode()
-
-  override fun getShadowNodeClass() = SafeAreaViewShadowNode::class.java
-
   @ReactProp(name = "mode")
   fun setMode(view: SafeAreaView, mode: String?) {
     when (mode) {
diff --git a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewShadowNode.kt b/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewShadowNode.kt
deleted file mode 100644
index 939950a..0000000
--- a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewShadowNode.kt
+++ /dev/null
@@ -1,170 +0,0 @@
-package com.th3rdwave.safeareacontext
-
-import com.facebook.react.bridge.Dynamic
-import com.facebook.react.bridge.ReadableType
-import com.facebook.react.uimanager.*
-import com.facebook.react.uimanager.annotations.ReactPropGroup
-import kotlin.math.max
-
-class SafeAreaViewShadowNode : LayoutShadowNode() {
-  private var mLocalData: SafeAreaViewLocalData? = null
-  private val mPaddings: FloatArray = FloatArray(ViewProps.PADDING_MARGIN_SPACING_TYPES.size)
-  private val mMargins: FloatArray = FloatArray(ViewProps.PADDING_MARGIN_SPACING_TYPES.size)
-  private var mNeedsUpdate = false
-
-  init {
-    for (i in ViewProps.PADDING_MARGIN_SPACING_TYPES.indices) {
-      mPaddings[i] = Float.NaN
-      mMargins[i] = Float.NaN
-    }
-  }
-
-  private fun updateInsets() {
-    val localData = mLocalData ?: return
-    var top = 0f
-    var right = 0f
-    var bottom = 0f
-    var left = 0f
-    val meta = if (localData.mode == SafeAreaViewMode.PADDING) mPaddings else mMargins
-    val allEdges = meta[Spacing.ALL]
-    if (!java.lang.Float.isNaN(allEdges)) {
-      top = allEdges
-      right = allEdges
-      bottom = allEdges
-      left = allEdges
-    }
-    val verticalEdges = meta[Spacing.VERTICAL]
-    if (!java.lang.Float.isNaN(verticalEdges)) {
-      top = verticalEdges
-      bottom = verticalEdges
-    }
-    val horizontalEdges = meta[Spacing.HORIZONTAL]
-    if (!java.lang.Float.isNaN(horizontalEdges)) {
-      right = horizontalEdges
-      left = horizontalEdges
-    }
-    val topEdge = meta[Spacing.TOP]
-    if (!java.lang.Float.isNaN(topEdge)) {
-      top = topEdge
-    }
-    val rightEdge = meta[Spacing.RIGHT]
-    if (!java.lang.Float.isNaN(rightEdge)) {
-      right = rightEdge
-    }
-    val bottomEdge = meta[Spacing.BOTTOM]
-    if (!java.lang.Float.isNaN(bottomEdge)) {
-      bottom = bottomEdge
-    }
-    val leftEdge = meta[Spacing.LEFT]
-    if (!java.lang.Float.isNaN(leftEdge)) {
-      left = leftEdge
-    }
-    top = PixelUtil.toPixelFromDIP(top)
-    right = PixelUtil.toPixelFromDIP(right)
-    bottom = PixelUtil.toPixelFromDIP(bottom)
-    left = PixelUtil.toPixelFromDIP(left)
-    val edges = localData.edges
-    val insets = localData.insets
-    if (localData.mode == SafeAreaViewMode.PADDING) {
-      super.setPadding(Spacing.TOP, getEdgeValue(edges.top, insets.top, top))
-      super.setPadding(Spacing.RIGHT, getEdgeValue(edges.right, insets.right, right))
-      super.setPadding(Spacing.BOTTOM, getEdgeValue(edges.bottom, insets.bottom, bottom))
-      super.setPadding(Spacing.LEFT, getEdgeValue(edges.left, insets.left, left))
-    } else {
-      super.setMargin(Spacing.TOP, getEdgeValue(edges.top, insets.top, top))
-      super.setMargin(Spacing.RIGHT, getEdgeValue(edges.right, insets.right, right))
-      super.setMargin(Spacing.BOTTOM, getEdgeValue(edges.bottom, insets.bottom, bottom))
-      super.setMargin(Spacing.LEFT, getEdgeValue(edges.left, insets.left, left))
-    }
-  }
-
-  private fun getEdgeValue(
-      edgeMode: SafeAreaViewEdgeModes,
-      insetValue: Float,
-      edgeValue: Float
-  ): Float {
-    if (edgeMode == SafeAreaViewEdgeModes.OFF) {
-      return edgeValue
-    } else if (edgeMode == SafeAreaViewEdgeModes.MAXIMUM) {
-      return max(insetValue, edgeValue)
-    } else {
-      return insetValue + edgeValue
-    }
-  }
-
-  private fun resetInsets(mode: SafeAreaViewMode) {
-    if (mode == SafeAreaViewMode.PADDING) {
-      super.setPadding(Spacing.TOP, mPaddings[Spacing.TOP])
-      super.setPadding(Spacing.RIGHT, mPaddings[Spacing.RIGHT])
-      super.setPadding(Spacing.BOTTOM, mPaddings[Spacing.BOTTOM])
-      super.setPadding(Spacing.LEFT, mPaddings[Spacing.LEFT])
-    } else {
-      super.setMargin(Spacing.TOP, mMargins[Spacing.TOP])
-      super.setMargin(Spacing.RIGHT, mMargins[Spacing.RIGHT])
-      super.setMargin(Spacing.BOTTOM, mMargins[Spacing.BOTTOM])
-      super.setMargin(Spacing.LEFT, mMargins[Spacing.LEFT])
-    }
-    markUpdated()
-  }
-
-  override fun onBeforeLayout(nativeViewHierarchyOptimizer: NativeViewHierarchyOptimizer) {
-    if (mNeedsUpdate) {
-      mNeedsUpdate = false
-      updateInsets()
-    }
-  }
-
-  override fun setLocalData(data: Any) {
-    if (data !is SafeAreaViewLocalData) {
-      return
-    }
-    val localData = mLocalData
-    if (localData != null && localData.mode != data.mode) {
-      resetInsets(localData.mode)
-    }
-    mLocalData = data
-    mNeedsUpdate = false
-    updateInsets()
-  }
-
-  // Names needs to reflect exact order in LayoutShadowNode.java
-  @ReactPropGroup(
-      names =
-          [
-              ViewProps.PADDING,
-              ViewProps.PADDING_VERTICAL,
-              ViewProps.PADDING_HORIZONTAL,
-              ViewProps.PADDING_START,
-              ViewProps.PADDING_END,
-              ViewProps.PADDING_TOP,
-              ViewProps.PADDING_BOTTOM,
-              ViewProps.PADDING_LEFT,
-              ViewProps.PADDING_RIGHT])
-  override fun setPaddings(index: Int, padding: Dynamic) {
-    val spacingType = ViewProps.PADDING_MARGIN_SPACING_TYPES[index]
-    mPaddings[spacingType] =
-        if (padding.type == ReadableType.Number) padding.asDouble().toFloat() else Float.NaN
-    super.setPaddings(index, padding)
-    mNeedsUpdate = true
-  }
-
-  @ReactPropGroup(
-      names =
-          [
-              ViewProps.MARGIN,
-              ViewProps.MARGIN_VERTICAL,
-              ViewProps.MARGIN_HORIZONTAL,
-              ViewProps.MARGIN_START,
-              ViewProps.MARGIN_END,
-              ViewProps.MARGIN_TOP,
-              ViewProps.MARGIN_BOTTOM,
-              ViewProps.MARGIN_LEFT,
-              ViewProps.MARGIN_RIGHT])
-  override fun setMargins(index: Int, margin: Dynamic) {
-    val spacingType = ViewProps.PADDING_MARGIN_SPACING_TYPES[index]
-    mMargins[spacingType] =
-        if (margin.type == ReadableType.Number) margin.asDouble().toFloat() else Float.NaN
-    super.setMargins(index, margin)
-    mNeedsUpdate = true
-  }
-}
diff --git a/node_modules/react-native-safe-area-context/common/cpp/react/renderer/components/safeareacontext/RNCSafeAreaViewState.cpp b/node_modules/react-native-safe-area-context/common/cpp/react/renderer/components/safeareacontext/RNCSafeAreaViewState.cpp
index ea9327f..d3c8c3e 100644
--- a/node_modules/react-native-safe-area-context/common/cpp/react/renderer/components/safeareacontext/RNCSafeAreaViewState.cpp
+++ b/node_modules/react-native-safe-area-context/common/cpp/react/renderer/components/safeareacontext/RNCSafeAreaViewState.cpp
@@ -21,3 +21,5 @@ folly::dynamic RNCSafeAreaViewState::getDynamic() const {

 } // namespace react
 } // namespace facebook
+
+__attribute__((used, visibility("default"))) unsigned char nito_safe_area_state_anchor = 0;
diff --git a/node_modules/react-native-safe-area-context/ios/Fabric/RNCSafeAreaProviderComponentView.mm b/node_modules/react-native-safe-area-context/ios/Fabric/RNCSafeAreaProviderComponentView.mm
index f315ccd..95e911d 100644
--- a/node_modules/react-native-safe-area-context/ios/Fabric/RNCSafeAreaProviderComponentView.mm
+++ b/node_modules/react-native-safe-area-context/ios/Fabric/RNCSafeAreaProviderComponentView.mm
@@ -106,9 +106,9 @@ using namespace facebook::react;
         .insets =
             {
                 .top = safeAreaInsets.top,
-                .left = safeAreaInsets.left,
-                .bottom = safeAreaInsets.bottom,
                 .right = safeAreaInsets.right,
+                .bottom = safeAreaInsets.bottom,
+                .left = safeAreaInsets.left,
             },
         .frame =
             {
